### PR TITLE
run: service-to-service auth snippet

### PR DESCRIPTION
**Pull request by ahmetb**
Service to service authentication method sample for Cloud Run.

### CODE REVIEW

**Comment by ahmetb** [comment]
cc: @mcdonamp

**Comment by broady** [comment]
LGTM, will leave to @tbpg to merge

File: run/authentication/auth.go [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package authentication contains the authentication samples for Cloud Run.
+package authentication
+
+// [START service_to_service_auth]
```
Comment (tbpg) [inline run/authentication/auth.go]: The region tag should be in the samples tracker, which requires the region tags to start with the product name. So, this should be something like `run_service_to_service_auth`.

Comment (ahmetb) [reply run/authentication/auth.go]: Thanks! just created sample number 1692 in the system with that `tag`. Any ideas what's next?

**Comment by broady** [comment]
maybe we need a TokenSource for this?

similar to this, which also accepts an audience:
https://godoc.org/golang.org/x/oauth2/google#JWTAccessTokenSourceFromJSON

**Comment by ahmetb** [comment]
Yeah not a bad idea except

1. other languages won't have it so it'll create asymmetry (e.g. I can't look at Go example and code my own PHP)
2. introduces an extra dependency in user code.

**Pull request merged**